test/tests/modules/logs.test.js

```diff
@@ -0,0 +1,89 @@
+/* OpenSprinkler App
+ * Copyright (C) 2015 - present, Samer Albahra. All rights reserved.
+ *
+ * This file is part of the OpenSprinkler project <http://opensprinkler.com>.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/* global OSApp, assert, describe, it, beforeEach, afterEach */
+
+describe('OSApp.Logs', function () {
+
+	describe('clearLogs', function () {
+		var originalAreYouSure, originalIsOSPi, originalSendToOS, originalShowError;
+
+		beforeEach(function () {
+			// Mock OSApp.UIDom.areYouSure
+			originalAreYouSure = OSApp.UIDom.areYouSure;
+			OSApp.UIDom.areYouSure = function (message, title, callback) {
+				callback(); // Simulate confirming the action
+			};
+
+			// Mock OSApp.Firmware.isOSPi
+			originalIsOSPi = OSApp.Firmware.isOSPi;
+			OSApp.Firmware.isOSPi = function () {
+				return false; // or true, depending on what you want to test
+			};
+
+			// Mock OSApp.Firmware.sendToOS
+			originalSendToOS = OSApp.Firmware.sendToOS;
+			OSApp.Firmware.sendToOS = function () {
+				return {
+					done: function (callback) {
+						callback(); // Simulate successful sendToOS
+						return this;
+					}
+				};
+			};
+
+			// Mock OSApp.Errors.showError
+			originalShowError = OSApp.Errors.showError;
```

Comment:
I don't think we should be mocking all of this (same comments so will stop making these).